tests/Microsoft.Azure.Batch.SoftwareEntitlement.Common.Tests/SpecifiableTests.cs

```diff
@@ -0,0 +1,88 @@
+using FluentAssertions;
+using Xunit;
+
+namespace Microsoft.Azure.Batch.SoftwareEntitlement.Common.Tests
+{
+    public class SpecifiableTests
+    {
+        private readonly Specifiable<string> _defaultOfString = default;
+
+        public class Construction : SpecifiableTests
+        {
+            [Fact]
+            public void CreatedAsParameterlessNew_IsNotSpecified()
+            {
+                var result = new Specifiable<string>();
+                result.IsSpecified.Should().BeFalse();
+            }
+
+            [Fact]
+            public void CreatedAsDefault_IsNotSpecified()
+            {
+                var result = _defaultOfString;
+                result.IsSpecified.Should().BeFalse();
+            }
+
+            [Fact]
+            public void CreatedWithValue_IsSpecified()
+            {
+                const string value = "test";
+                var result = new Specifiable<string>(value);
+                result.IsSpecified.Should().BeTrue();
+            }
+
+            [Fact]
+            public void CreatedWithNull_IsSpecified()
+            {
+                const string value = null;
+                var result = new Specifiable<string>(value);
+                result.IsSpecified.Should().BeTrue();
+            }
+
+            [Fact]
+            public void CreatedBySpecifyAsValue_IsSpecified()
+            {
+                const string value = "test";
+                var result = Specify.As(value);
```

Comment:
I'd separate out the tests for `Specify.As()`, they're not really about testing the constructor.